tests/SmartpayTest.php

```diff
@@ -8,52 +8,38 @@
 
 final class SmartpayTest extends TestCase
 {
-    public function testApiUrlAccessors()
+    public function set_up()
     {
-        Smartpay::setApiUrl(null);
-        static::assertSame('https://api.smartpay.co/v1', Smartpay::getApiUrl());
-
-        Smartpay::setApiUrl('https://api.smartpay.co/v2');
-        static::assertSame('https://api.smartpay.co/v2', Smartpay::getApiUrl());
-    }
-
-    public function testCheckoutUrlAccessors()
-    {
-        static::assertSame('https://checkout.smartpay.co', Smartpay::getCheckoutUrl());
-
-        Smartpay::setCheckoutUrl(null);
-        static::assertSame('https://checkout.smartpay.co', Smartpay::getCheckoutUrl());
-
-        Smartpay::setCheckoutUrl('https://checkout.smartpay.co/v1');
-        static::assertSame('https://checkout.smartpay.co/v1', Smartpay::getCheckoutUrl());
+        putenv('SMARTPAY_PUBLIC_KEY');
+        putenv('SMARTPAY_SECRET_KEY');
+        putenv('SMARTPAY_API_PREFIX');
+        parent::set_up();
     }
 
-    public function testPostTimeoutAccessors()
+    public function testConstructorWithoutArguments()
     {
-        static::assertSame(30, Smartpay::getPostTimeout());
-
-        Smartpay::setPostTimeout(null);
-        static::assertSame(30, Smartpay::getPostTimeout());
-
-        Smartpay::setPostTimeout(100);
-        static::assertSame(100, Smartpay::getPostTimeout());
+        $smartpay = new Smartpay();
+        $this->assertEquals('', $smartpay->getPublicKey());
+        $this->assertEquals('', $smartpay->getSecretKey());
+        $this->assertEquals(Smartpay::DEFAULT_API_URL, $smartpay->getApiUrl());
     }
 
-    public function testPublicKeyAccessors()
+    public function testConstructorWithEnv()
     {
-        Smartpay::setPublicKey(null);
-        static::assertNull(Smartpay::getPublicKey());
-
-        Smartpay::setPublicKey('pk_test_123');
-        static::assertSame('pk_test_123', Smartpay::getPublicKey());
+        putenv('SMARTPAY_PUBLIC_KEY=public_key');
+        putenv('SMARTPAY_SECRET_KEY=secret_key');
+        putenv('SMARTPAY_API_PREFIX=api_url');
+        $smartpay = new Smartpay();
+        $this->assertEquals('public_key', $smartpay->getPublicKey());
+        $this->assertEquals('secret_key', $smartpay->getSecretKey());
+        $this->assertEquals('api_url', $smartpay->getApiUrl());
     }
 
-    public function testSecretKeyAccessors()
+    public function testConstructor()
     {
-        Smartpay::setSecretKey(null);
-        static::assertNull(Smartpay::getSecretKey());
-
-        Smartpay::setSecretKey('sk_test_123');
-        static::assertSame('sk_test_123', Smartpay::getSecretKey());
+        $smartpay = new Smartpay('sec', 'pub');
```

Comment:
Can we have a test case to have both ENV_VAR and the customer key as arguments when calling the constructor, and make sure the lib will use the later one?

Reply:
Added